feat: Anclap and Stellar nodes

lucasmpe opened the pull request:
Summary:

This pr moves the Anclap and Stellar nodes to the nodes-base package.

In turn, it replaces the original CI workflows with a single workflow that executes the tests of the nodes-base package.

lucasmpe committed:
Move anclap and stellar nodes to nodes-base package

- Add missing types
- Fix import types

lucasmpe committed:
Replace the original ci workflows with a customized one for the nodes-base package

lucasmpe committed:
Attempt to fix ci workflow

Add default master branch

Genebson commented on packages/nodes-base/nodes/Anclap/actions/transactions/withdraw/description.ts:
```diff
@@ -136,13 +133,12 @@ export const withdrawDescription: INodeProperties[] = [
 		default: '',
 		description:
-			'Type of memo. One of text, id or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of id.',
+			'Type of memo. One of text, ID or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of ID.',
```
```suggestion
			'Type of memo. One of text, ID or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of ID',
```

Genebson commented on packages/nodes-base/nodes/Anclap/actions/transactions/withdraw/description.ts:
```diff
@@ -248,6 +238,6 @@ export const withdrawDescription: INodeProperties[] = [
 		default: '',
 		description:
-			'The type of the refund_memo. Can be id, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified.',
+			'The type of the refund_memo. Can be ID, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified.',
```
```suggestion
			'The type of the refund_memo. Can be ID, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified',
```

Genebson commented on packages/nodes-base/nodes/Anclap/actions/transactions/withdrawExchange/description.ts:
```diff
@@ -136,13 +134,12 @@ export const withdrawExchangeDescription: INodeProperties[] = [
 		default: '',
 		description:
-			"The id returned from a SEP-38 POST /quote response. If this parameter is provided and the Stellar transaction used to send the asset to the Anchor has a created_at timestamp earlier than the quote's expires_at attribute, the Anchor should respect the conversion rate agreed in that quote. If the values of destination_asset, source_asset and amount conflict with the ones used to create the SEP- ...
+			"The ID returned from a SEP-38 POST /quote response. If this parameter is provided and the Stellar transaction used to send the asset to the Anchor has a created_at timestamp earlier than the quote's expires_at attribute, the Anchor should respect the conversion rate agreed in that quote. If the values of destination_asset, source_asset and amount conflict with the ones used to create the SEP- ...
```
```suggestion
			"The ID returned from a SEP-38 POST /quote response. If this parameter is provided and the Stellar transaction used to send the asset to the Anchor has a created_at timestamp earlier than the quote's expires_at attribute, the Anchor should respect the conversion rate agreed in that quote. If the values of destination_asset, source_asset and amount conflict with the ones used to create the SEP-38 quote, this request should be rejected with a 400.",
```

Genebson commented on packages/nodes-base/nodes/Anclap/actions/transactions/withdrawExchange/description.ts:
```diff
@@ -168,13 +164,12 @@ export const withdrawExchangeDescription: INodeProperties[] = [
 		default: '',
 		description:
-			'Type of memo. One of text, id or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of id.',
+			'Type of memo. One of text, ID or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of ID.',
```
```suggestion
			'Type of memo. One of text, ID or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of ID',
```

Genebson commented on packages/nodes-base/nodes/Anclap/actions/transactions/withdrawExchange/description.ts:
```diff
@@ -280,6 +269,6 @@ export const withdrawExchangeDescription: INodeProperties[] = [
 		default: '',
 		description:
-			'The type of the refund_memo. Can be id, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified.',
+			'The type of the refund_memo. Can be ID, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified.',
```
```suggestion
			'The type of the refund_memo. Can be ID, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified.',
```
```suggestion
			'The type of the refund_memo. Can be ID, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified',
```

Genebson commented on packages/nodes-base/nodes/Anclap/actions/transactions/withdrawInteractive/description.ts:
```diff
@@ -136,13 +133,12 @@ export const withdrawInteractiveDescription: INodeProperties[] = [
 		default: '',
 		description:
-			'Type of memo. One of text, id or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of id.',
+			'Type of memo. One of text, ID or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of ID.',
```
```suggestion
			'Type of memo. One of text, ID or hash. Deprecated because memos used to identify users of the same Stellar account should always be of type of ID',
```

Genebson commented on packages/nodes-base/nodes/Anclap/actions/transactions/withdrawInteractive/description.ts:
```diff
@@ -248,6 +238,6 @@ export const withdrawInteractiveDescription: INodeProperties[] = [
 		default: '',
 		description:
-			'The type of the refund_memo. Can be id, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified.',
+			'The type of the refund_memo. Can be ID, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified.',
```
```suggestion
			'The type of the refund_memo. Can be ID, text, or hash. See the memos documentation for more information. If specified, refund_memo must also be specified',
```

Genebson commented on packages/nodes-base/nodes/Anclap/transport/SEP12.ts:
```diff
@@ -122,11 +125,19 @@ export default class SEP12 {
 		try {
-			const fileId = await axios.post(`${toml.KYC_SERVER}/customer/files`, request, {
-				headers: { Authorization: `Bearer ${this.token}` },
+			const fileId = await axios.post(`${toml.KYC_SERVER}/customer/files`, formData, {
```
I think you can destructure data here. Thoughts?

Genebson commented on packages/nodes-base/nodes/Anclap/transport/helpers.ts:
```diff
@@ -1,9 +1,16 @@
 import { NetworkPassphrase } from './enums/networkPassphrase';
 
+function extractHomeDomain(anchorUrl: string): string {
+	const url = new URL(anchorUrl);
+
+	return url.hostname;
```
```suggestion
	const { hostname } = new URL(anchorUrl);
	return hostname;
```

Genebson commented on packages/nodes-base/nodes/Stellar/actions/entities/ILiquidtyPoolPrice.ts:
```diff
@@ -0,0 +1,4 @@
+export interface ILiquidtyPoolPrice {
+	n: string;
+	d: string;
```
What does this mean?

lucasmpe replied:
This interface is not used. I will remove it.

Genebson commented on packages/nodes-base/nodes/Stellar/actions/entities/IPriceFraction.ts:
```diff
@@ -0,0 +1,4 @@
+export interface IFraction {
+	n: string;
+	d: string;
```
Same here.

lucasmpe replied:
These mean numerator and denominator. I will make them explicit.

lucasmpe committed:
Refactor Anclap and Stellar nodes

- Fix property descriptions in Anclap node
- Add object destructuring in Anclap node
- Fix interface property names in Stellar node
- Remove unused interface in Stellar node
